src/jit/morph.cpp

```diff
@@ -6188,7 +6188,6 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
         tree->SetOper(GT_IND);
         tree->gtOp.gtOp1 = addr;
 
-        tree->gtFlags &= (~GTF_EXCEPT | addr->gtFlags);
```

Comment:
why does this go away?
` tree->gtFlags &= addr->gtFlags;`

Reply:
`~GTF_EXCEPT | addr->gtFlags` has all bits set except, possibly for `GTF_EXCEPT` bit. `SetIndirExceptionFlags` will set that bit correctly.